back/src/test/java/com/ubikloadpack/jmeter/ulp/observability/server/ULPObservabilityMetricServletTest.java

```diff
@@ -115,6 +120,27 @@ public void whenTotalLabelCouldContainsSpacesExpectMetricsGeneratedAndLogged() t
 		assertSamplersCountEveryPeriods(actualMetrics, totalLabelOpenMetric, 2, sampleNameOpenMetric, 1); // totalLabel's count becomes 2
 	}
 	
+	@Test
+	@DisplayName("When a sample fails due to an error, expect the occurred error is reported as a top error")
+	public void whenErrorOccurredOnSampleExpectItOnTopErrors() throws Exception {

```

Comment:
Can you add a test, so that we have at least 5 different type of error with different occurences and check if result is fine ?